### PR TITLE
add public key validator

### PR DESCRIPTION
**Pull request by rogachevsn55**

### CODE REVIEW

File: src/main/java/jp/co/soramitsu/sora/didresolver/dto/Proof.java [hunk 1]
```diff
@@ -4,32 +4,32 @@
...
-    private String creator;
+  @NotBlank
+  @KeyConstraint
+  private String creator;
 
-    private String signatureValueBase58;
+  private String signatureValueBase58;
 
-    private String signatureValueHex;
+  private String signatureValueHex;
```
Comment (Warchant) [inline src/main/java/jp/co/soramitsu/sora/didresolver/dto/Proof.java]: just signatureValue, remove base58

File: src/main/java/jp/co/soramitsu/sora/didresolver/validation/KeyValidator.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class KeyValidator implements ConstraintValidator<KeyConstraint, String> {
+
+  private static final String KEY_FORMAT_PATTERN = ".*" + Consts.DID_URI_DETERMINATOR + "keys-\\d";
+
+  @Override
+  public boolean isValid(String publicKey, ConstraintValidatorContext context) {
+    log.debug("validation format of public key - {}", publicKey);
```
Comment (Warchant) [inline src/main/java/jp/co/soramitsu/sora/didresolver/validation/KeyValidator.java]: did is uri

why you don't use java.net.URI? it has .getFragment or something like that

File: src/main/java/jp/co/soramitsu/sora/didresolver/dto/Proof.java (hunk 1 above)
Comment (rogachevsn55) [reply src/main/java/jp/co/soramitsu/sora/didresolver/dto/Proof.java]: these changes in another PR

File: src/main/java/jp/co/soramitsu/sora/didresolver/validation/KeyValidator.java (hunk 2 above)
Comment (rogachevsn55) [reply src/main/java/jp/co/soramitsu/sora/didresolver/validation/KeyValidator.java]: yes, but for DID validation you still have to parse it as a string. so the controversial question is whether it should be used and if you use then change the type in the DTO itself or only in the validator

Comment (Warchant) [reply src/main/java/jp/co/soramitsu/sora/didresolver/validation/KeyValidator.java]: @rogachevsn55 this has to be discussed, I think, whether we change DTO or not

**Pull request merged**